doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -983,3 +984,11 @@ on success. It is stronger than the tactic ``fold``.
    .. coqtop:: none
 
       Abort.
+
+.. note::
+   The name `fail` may refer both to the tactic :tac:`fail` and to
+   rewriting strategy `fail`. Similarly `;` may refer both to a
+   sequence of tactics and to a sequence of rewriting strategies. For
+   instance, `rewrite_strat s; fail` stands for `rewrite_strat (s; fail)`
+   by default. Use `(rewrite_strat s); fail` to mean a sequence of
+   tactics ending with the tactic :tac:`fail`.
```

Comment:
**_Would you put this before the example (more consistent with rest of the doc)?_**

```suggestion
.. note::
   ';' is used to separate sequences of tactics as well as
   sequences of rewriting strategies.
   `rewrite_strat s; fail` is interpreted as `rewrite_strat (s; fail)`,
   in which `fail` is a rewriting strategy.
   Use `(rewrite_strat s); fail` to make `fail` a tactic.
   `rewrite_strat s; apply I` gives a syntax error (`apply` is not
   a valid rewrite strategy).
```

Reply:
> Would you put this before the example (more consistent with rest of the doc)?

Done.

For the suggestion, I don't know. Why not, but preferably not starting the sentence with a semicolon. So, maybe "The symbol ';' is used to separate sequences...".

Reply:
Good point.  You could just say "Semicolons are used ..."